pkg/nftables/firewall.go

```diff
@@ -27,6 +29,13 @@ const (
 //go:embed *.tpl
 var templates embed.FS
 
+//go:generate mockgen -destination=./mocks/mock_fqdncache.go -package=mocks . FQDNCache
+type FQDNCache interface {
+	UpdateDNSServerAddr(addr string)
+	GetSetsForRendering(fqdns []firewallv1.FQDNSelector) (result []firewallv1.IPSet)
+	GetSetsForFQDN(fqdn firewallv1.FQDNSelector, update bool) (result []firewallv1.IPSet)
+}
```

Comment:
Sometimes interfaces make code harder to read. Maybe it's easier and also "okay-ish" to pass it the DNS Proxy? I guess you wanted to clearly isolate some functions from the proxy and also make the code testable, but I am not sure if the extra interface and mockgen is worth it.